includes/classes/SiteAutomation/EndPoints.php

```diff
@@ -0,0 +1,343 @@
+			$post_data_updated = $this->get_post_details( $override_post_ID, $rules );
+			$curated_posts[]   = (object) array_merge( (array) $post_data, (array) $post_data_updated );
+
+			return $curated_posts;
+		}
+
+		foreach ( $curated_posts as $index => $curated_post ) {
+			$post_data_updated       = $this->get_post_details( $curated_post->ID, $rules );
+			$curated_posts[ $index ] = (object) array_merge( (array) $curated_post, (array) $post_data_updated );
+		}
+
+		return $curated_posts;
+	}
+
+	/**
+	 * Get the post data with prepared items to be rendered to the inner blocks.
+	 *
+	 * @param int   $post_ID Post ID.
+	 * @param array $rules array of the parent block settings.
+	 *
+	 * @return \stdClass Prepared post data.
+	 */
+	public function get_post_details( $post_ID, $rules ) {
+		$post_data = new \stdClass();
+
+		$post_data->postLink = get_the_permalink( $post_ID );
+		if ( $rules['display_featured_image'] ) {
+			$post_data->featuredImage = get_the_post_thumbnail( $post_ID );
+		}
+		if ( $rules['display_author'] ) {
+			$author_id             = get_post_field( 'post_author', $post_ID );
+			$author_display_name   = get_the_author_meta( 'display_name', $author_id );
+			$byline                = sprintf( __( 'by %s', 'sophi-wp' ), $author_display_name );
+			$post_data->postAuthor = $byline;
+		}
+		if ( $rules['display_post_date'] ) {
+			$post_data->postDate  = get_the_date( '', $post_ID );
+			$post_data->postDateC = get_the_date( 'c', $post_ID );
+		}
+		if ( $rules['display_post_excerpt'] ) {
+			$post_data->post_excerpt = wp_kses_post( get_the_excerpt( $post_ID ) );
+		}
+
+		return $post_data;
+	}
+
+	/**
+	 * Get the query params to get the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return array
+	 */
+	public function site_automation_params() {
+		$params['pageName'] = array(
+			'description'       => __( 'Name of the page.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['widgetName'] = array(
+			'description'       => __( 'Name of the widget.', 'sophi-wp' ),
+			'required'          => true,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		/**
+		 * Filters the get sophi posts query params.
+		 *
+		 * @param array $params Query params.
+		 *
+		 * @since 1.3.0
+		 */
+		return apply_filters( 'site_automation_params', $params );
+	}
+
+	/**
+	 * Update the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @param  WP_REST_Request $request The request.
+	 * @return mixed
+	 */
+	public function site_automation_override( $request ) {
+		$current_user = wp_get_current_user();
+
+		if ( $current_user->exists() ) {
+			$user_email = $current_user->user_email;
+		} else {
+			return new \WP_Error( 401, __( 'Unauthorised user, please log in.', 'sophi-wp' ) );
```

Comment:
I mention this above but probably worth moving this into a `permission_callback` instead of having it here. Also, we only check if a user is currently logged-in/exists, do we need to lock this down to specific user roles?

Reply:
I think yes, but it depends on what is the specific requirement here. If the Gutenberg block should be accessible to all users who can edit the post, then we are good, but if not, we need which roles should be allowed/not-allowed. cc: @jeffpaul